styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -681,11 +683,17 @@ private static DockerRunner createDockerRunner(
       LOG.info("Creating LocalDockerRunner");
       return closer.register(DockerRunner.local(scheduler, stateManager));
     } else {
+      final String styxEnvironment;
+      if (!config.hasPath(STYX_ENVIRONMENT)) {
```

Comment:
Would it be better perhaps to not have a default environment? I.e. fail if `styx.environment` is not set.

Reply:
Afact this is how we handle `styx.mode`.

Reply:
Is it? I thought we just created different classes depending on `styx.mode`.

Reply:
Ah ok it fails by getting non-configured value.